dom/cli/init.py

```
@@ -0,0 +1,296 @@
+import os
+import typer
+import datetime
+from rich.panel import Panel
+from rich.progress import Progress, SpinnerColumn, TextColumn
+from rich.prompt import Prompt, Confirm
+from rich.table import Table
+
+from . import console
+
+init_command = typer.Typer()
+
+INFRA_TEMPLATE = """infra:
+  port: {port}
+  judges: {judges}
+  password: "{password}"
+"""
+
+CONTEST_TEMPLATE = """
+contests:
+  - name: "{name}"
+    shortname: "{shortname}"
+    start_time: "{start_time}"
+    duration: "{duration}"
+    penalty_time: {penalty_time}
+    allow_submit: {allow_submit}
+
+    problems:
+      from: "problems.yaml"
+
+    teams:
+      from: "{teams}"
+      delimiter: ','
+      rows: "2-50"
+      name: "$2"
+"""
+
+PROBLEMS_TEMPLATE = """- archive: "{archive}"
+  platform: "{platform}"
+  color: "{color}"
+"""
+
+def format_datetime(date_str):
+    """Convert datetime string to ISO 8601 format with timezone"""
+    try:
+        dt = datetime.datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S")
+        return dt.strftime("%Y-%m-%dT%H:%M:%S+00:00")
+    except ValueError:
+        return date_str
+
+def format_duration(duration_str):
+    """Format duration to include milliseconds"""
+    if "." not in duration_str:
+        return duration_str + ".000"
+    return duration_str
+
+@init_command.callback(invoke_without_command=True)
+def callback():
+    """
+    Initialize the DOMjudge configuration files with an interactive wizard.
+    """
```

Can you please raise an exception if a `dom-judge.yaml` or `dom-judge.yml` file already exists? This will help prevent accidental overwrites.

To keep things clean, please move the existence check to a new helper function in `dom/utils/cli.py` so we can centralize this logic and reuse it where needed.

---

Optionally, you can add a keyword argument `--overwrite` that can bypass this check